jraph/_src/utils.py

```diff
@@ -1122,3 +1123,33 @@ def wrapper(graph: gn_graph.GraphsTuple) -> gn_graph.GraphsTuple:
     return zero_out_padding(graph_net(graph))
 
   return wrapper
+
+
+def from_scipy_sparse_matrix(A: scipy.sparse.spmatrix):
+  """Creates a `jraph.GraphsTuple` from a scipy sparse matrix.
+  
+  Args:
+    A: A SciPy sparse matrix.
+    
+  Returns:
+    A `jraph.GraphsTuple` graph based on the scipy sparse matrix `A`.
+  """
+  A = A.tocoo()
+  n_node = jnp.asarray([A.shape[1]])
+  n_edge = jnp.asarray([jnp.sum(A.data)])
+  
+  senders = []
+  receivers = []
+  for idx, (sender_idx, receiver_idx) in enumerate(zip(A.row, A.col)):
```

Comment:
I would prefer using np.repeat here, it will be a lot lot faster.

Reply:
This suggestion is amazing, I didn't know we can do this!